api/v2/serializers/details/quota_request.py

```diff
@@ -0,0 +1,161 @@
+        serializer = IdentitySummarySerializer(identity, context=self.context)
+        return serializer.data
+
+    def to_internal_value(self, data):
+        queryset = self.get_queryset()
+        if isinstance(data, dict):
+            identity = data.get("id", None)
+        else:
+            identity = data
+        try:
+            return queryset.get(id=identity)
+        except:
+            raise exceptions.ValidationError(
+                "Identity with id '%s' does not exist."
+                % identity
+            )
+
+class QuotaRelatedField(serializers.PrimaryKeyRelatedField):
+
+    def get_queryset(self):
+        return Quota.objects.all()
+
+    def to_representation(self, value):
+        if value.pk is None:
+            return None
+
+        quota = Quota.objects.get(pk=value.pk)
+        serializer = QuotaSummarySerializer(quota, context=self.context)
+        return serializer.data
+
+    def to_internal_value(self, data):
+        queryset = self.get_queryset()
+        if isinstance(data, dict):
+            identity = data.get("id", None)
+        else:
+            identity = data
+        try:
+            return queryset.get(id=identity)
+        except:
+            raise exceptions.ValidationError(
+                "Quota with id '%s' does not exist."
+                % identity
+            )
+
+class StatusTypeRelatedField(serializers.PrimaryKeyRelatedField):
+
+    def get_queryset(self):
+        return StatusType.objects.all()
+
+    def to_representation(self, value):
+        status_type = StatusType.objects.get(pk=value.pk)
+        serializer = StatusTypeSummarySerializer(status_type, context=self.context)
+        return serializer.data
+
+    def to_internal_value(self, data):
+        queryset = self.get_queryset()
+        if isinstance(data, dict):
+            identity = data.get("id", None)
+        else:
+            identity = data
+
+        try:
+            return queryset.get(id=identity)
+        except:
+            raise exceptions.ValidationError(
+                "StatusType with id '%s' does not exist."
+                % identity
+            )
+
+class QuotaRequestSerializer(serializers.HyperlinkedModelSerializer):
+    uuid = serializers.CharField(read_only=True)
+    created_by = UserRelatedField(read_only=True)
+    user = UserSummarySerializer(source='membership.identity.created_by', read_only=True)
+    identity = IdentityRelatedField(source='membership.identity',
+                                    queryset=Identity.objects.none())
+    provider = ProviderSummarySerializer(source='membership.identity.provider', read_only=True)
+    status = StatusTypeRelatedField(queryset=StatusType.objects.none(),
+                                    allow_null=True,
+                                    required=False)
+    quota = QuotaRelatedField(queryset=Quota.objects.all(),
+                              allow_null=True,
+                              required=True)
```

Comment:
I'm wondering why you subclassed PrimaryKeyRelatedField here, since its a new API model/view/serializer, you could have nested the serializers, and then override the 'create/update' fields in the serializer to handle nested saving..